File: leetcode_vs/14.最长公共前缀.py
#
# @lc app=leetcode.cn id=14 lang=python3
#
# [14] 最长公共前缀
#

# @lc code=start
import sys


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        row = len(strs)
        if row == 0:
            return ""
        if row == 1:
            return strs[0]
        col = sys.maxsize
        length = 0
        flag = True
        for s in strs:
            col = min(col, len(s))

        for j in range(col):
            same = strs[0][j]
            for i in range(row):
                if strs[i][j] != same:
                    flag = False
                    break
            if flag:
                length += 1
            else:
                break
        return strs[0][0:length] if length else ""
# @lc code=end

File: leetcode_vs/14.最长公共前缀.java
/*
 * @lc app=leetcode.cn id=14 lang=java
 *
 * [14] 最长公共前缀
 */

// @lc code=start
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int n = strs.length;
        if (n == 0)
            return "";
        StringBuffer res = new StringBuffer();

        for (int j = 0; j < strs[0].length(); j++) {
            char c = strs[0].charAt(j);
            for (int i = 1; i < n; i++) {
                
                if (strs[i].length()<=j ||c != strs[i].charAt(j)) {
                    return res.toString();
                }
                
            }
            res.append(c);
        }
        return res.toString();
    }
}
// @lc code=end
